Sources/Target/Common/ProcessBase.cpp

```diff
@@ -288,6 +284,26 @@ ErrorCode ProcessBase::afterResume() {
   return kSuccess;
 }
 
+SoftwareBreakpointManager *ProcessBase::softwareBreakpointManager() const {
+#if defined(ARCH_ARM) || defined(ARCH_ARM64)
+  typedef Architecture::ARM::SoftwareBreakpointManager SWBPMType;
+#elif defined(ARCH_X86) || defined(ARCH_X86_64)
+  typedef Architecture::X86::SoftwareBreakpointManager SWBPMType;
+#endif
+  static SWBPMType softwareBreakpointManager(const_cast<ProcessBase *>(this));
```

Comment:
Not that it matters here, but the cool thing with the static local variables is that the initialization is inherently thread-safe.